src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java

```diff
@@ -0,0 +1,153 @@
+    @JsonIgnore
+    private PhysicalPortData data;
+
+    /**
+     * No args constructor for deserialization purpose.
+     */
+    private PhysicalPort() {
+        data = new PhysicalPortDataImpl();
+    }
+
+    /**
+     * Cloning constructor which performs deep copy of the entity.
+     *
+     * @param entityToClone the entity to copy entity data from.
+     */
+    public PhysicalPort(@NonNull PhysicalPort entityToClone) {
+        data = PhysicalPortCloner.INSTANCE.deepCopy(entityToClone.getData());
+    }
+
+    @Builder
+    public PhysicalPort(@NonNull SwitchId switchId, int portNumber, @NonNull LagLogicalPort lagLogicalPort) {
+        data = new PhysicalPortDataImpl(switchId, portNumber, lagLogicalPort);
+    }
+
+    public PhysicalPort(@NonNull PhysicalPort.PhysicalPortData data) {
+        this.data = data;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PhysicalPort that = (PhysicalPort) o;
+        return new EqualsBuilder()
+                .append(getPortNumber(), that.getPortNumber())
+                .append(getSwitchId(), that.getSwitchId())
+                .isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getPortNumber(), getSwitchId());
+    }
+
+    /**
+     * Defines persistable data of the PhysicalPort.
+     */
+    public interface PhysicalPortData {
+        SwitchId getSwitchId();
+
+        void setSwitchId(SwitchId switchId);
+
+        LagLogicalPort getLagLogicalPort();
+
+        void setLagLogicalPort(LagLogicalPort lagLogicalPort);
+
+        int getPortNumber();
+
+        void setPortNumber(int portNumber);
+    }
+
+    /**
+     * POJO implementation of PhysicalPortData.
+     */
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    static final class PhysicalPortDataImpl implements PhysicalPortData, Serializable {
+        private static final long serialVersionUID = 1L;
+        @NonNull SwitchId switchId;
+        int portNumber;
+
+        @ToString.Exclude
+        @EqualsAndHashCode.Exclude
+        @NonNull LagLogicalPort lagLogicalPort;
+    }
+
+    /**
+     * A cloner for PhysicalPort entity.
+     */
+    @Mapper
+    public interface PhysicalPortCloner {
+        PhysicalPortCloner INSTANCE = Mappers.getMapper(PhysicalPortCloner.class);
+
+        void copy(PhysicalPortData source, @MappingTarget PhysicalPortData target);
+
+        @Mapping(target = "lagLogicalPort", ignore = true)
+        void copyWithoutLagLogicalPort(PhysicalPortData source, @MappingTarget PhysicalPortData target);
+
+        /**
+         * Performs deep copy of entity data.
+         */
+        default PhysicalPortData deepCopy(PhysicalPortData source) {
+            PhysicalPortData result = new PhysicalPortDataImpl();
+            copyWithoutLagLogicalPort(source, result);
```

Comment:
Here was line 
```result.setLagLogicalPort(new LagLogicalPort(source.getLagLogicalPort()));```
but this line causes stack overflow.

I'm not sure should it be here or not.
@sergii-iakovenko Please help Dasha to understand how to copy DB objects

![maxresdefault](https://user-images.githubusercontent.com/8777565/127708668-8660223e-7c8d-4d0e-ab5d-752cb37612eb.jpg)

Reply:
Please, check how it's done for FlowMirrorPath / FlowMirrorPoints. In short, you can't do this with the same copy constructor of LagLogicalPort. Instead, you need another one which doesn't clone composite PhysicalPort(s).

Reply:
fixed